apis/src/components/organisms/header.rs

```diff
@@ -1,19 +1,62 @@
 use crate::components::atoms::next_game_button::NextGameButton;
+use crate::components::molecules::chat_and_controls::ChatAndControls;
 use crate::components::organisms::{
     darkmode_toggle::DarkModeToggle,
     dropdowns::{CommunityDropdown, MobileDropdown, UserDropdown},
 };
+use crate::providers::game_state::GameStateSignal;
 use crate::providers::AuthContext;
+use hive_lib::GameControl;
 use leptos::*;
 use leptos_router::use_location;
 use shared_types::TimeMode;
 
 #[derive(Clone)]
 pub struct Redirect(pub RwSignal<String>);
 
+#[derive(Clone)]
+pub struct ControlsSignal {
+    pub hidden: RwSignal<bool>,
+    pub notify: Signal<bool>,
+}
+
+#[derive(Clone, Copy)]
+pub struct MobileChat(pub RwSignal<bool>);
+
 #[component]
 pub fn Header(#[prop(optional)] extend_tw_classes: &'static str) -> impl IntoView {
     let auth_context = expect_context::<AuthContext>();
+    let gamestate = expect_context::<GameStateSignal>();
+    let user = move || match untrack(auth_context.user) {
+        Some(Ok(Some(user))) => Some(user),
+        _ => None,
+    };
+
+    let color = move || {
+        if let Some(user) = user() {
+            gamestate.user_color(user.id)
+        } else {
+            None
+        }
+    };
+    let has_gamecontrol = create_read_slice(gamestate.signal, move |gs| {
+        if let Some(color) = color() {
+            let opp_color = color.opposite_color();
+            matches!(
+                gs.game_control_pending,
+                Some(GameControl::TakebackRequest(color) | GameControl::DrawOffer(color)) if color == opp_color
+            )
+        } else {
+            false
+        }
+    });
+    let hide_controls = ControlsSignal {
+        hidden: RwSignal::new(true),
+        notify: has_gamecontrol,
+    };
+    let mobile_chat = MobileChat(RwSignal::new(false));
```

Comment:
What is `MobileChat`?

Reply:
Did I figure that out correctly, that it's more a general thing about mobile view instead of chat?